### PR TITLE
Add option to configurably set UserAgent string in Stackdriver exporter

### PR DESCRIPTION
**Pull request by james-bebbington**
Add option to configurably set UserAgent string in Stackdriver exporter. Also allow the user to specify `{{version}}` in this field and replace it with the app version.

### CODE REVIEW

File: exporter/stackdriverexporter/factory.go [hunk 1]
```diff
@@ -68,8 +68,8 @@ func createTraceExporter(
 // createMetricsExporter creates a metrics exporter based on this config.
 func createMetricsExporter(
 	_ context.Context,
-	_ component.ExporterCreateParams,
+	params component.ExporterCreateParams,
 	cfg configmodels.Exporter) (component.MetricsExporter, error) {
 	eCfg := cfg.(*Config)
-	return newStackdriverMetricsExporter(eCfg)
+	return newStackdriverMetricsExporter(eCfg, params.ApplicationStartInfo.Version)
```
Comment (nilebox) [inline exporter/stackdriverexporter/factory.go:74]: What about trace exporter? Do we need to add support for custom user agent to `opentelemetry-operations-go` first?

File: exporter/stackdriverexporter/stackdriver.go [hunk 2]
```diff
@@ -116,14 +117,24 @@ func newStackdriverMetricsExporter(cfg *Config) (component.MetricsExporter, erro
 		// Set DefaultMonitoringLabels to an empty map to avoid getting the "opencensus_task" label
 		DefaultMonitoringLabels: &stackdriver.Labels{},
 	}
+
+	userAgent := strings.ReplaceAll(cfg.UserAgent, "{{version}}", version)
 	if cfg.Endpoint != "" {
-		copts, err := generateClientOptions(cfg)
+		// WithGRPCConn option takes precedent over all other supplied options so need to provide user agent here as well
+		dialOpts := []grpc.DialOption{}
+		if userAgent != "" {
```
Comment (nilebox) [inline exporter/stackdriverexporter/stackdriver.go:125]: Given that OpenCensus exporter sets user agent to `"opencensus-go %s; stackdriver-exporter %s"` by default, we may want to override it with `"opentelemetry-collector-contrib {{version}}"` by default (i.e. even if it's not configured)?

**Review by nilebox — COMMENTED** [review]
Looks good overall.
See comments, and it seems that you will need to rebase after core dependency is updated to fix the build.

File: exporter/stackdriverexporter/factory.go (hunk 1 above)
Comment (james-bebbington) [reply exporter/stackdriverexporter/factory.go:78]: I've documented that this only applies to metrics for now. We'll need to update the operations exporter before applying this to traces. Tracked here: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/82

**Review by nilebox — APPROVED** [review]

**Pull request merged**